packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -117,8 +117,11 @@ class LandmarkManager {
   }
 
   public updateLandmark(landmark: Landmark) {
-    this.landmarks = this.landmarks.map((prevLandmark) => prevLandmark.ref.current === landmark.ref.current ? {...prevLandmark, ...landmark} : prevLandmark);
-    this.checkLabels(landmark.role);
+    let index = this.landmarks.findIndex(l => l.ref === landmark.ref);
+    if (index >= 0) {
+      this.landmarks[index] = {...landmark};
```

Comment:
this should still be responsible for getting the current landmark for updating
```suggestion
      this.landmarks[index] = {...this.landmarks[index], ...landmark};
```